r/src/arrow_types.h

```diff
@@ -90,6 +90,24 @@ class ConstReferenceSmartPtrInputParameter {
   const T* ptr;
 };
 
+template <typename T>
+class ConstReferenceVectorSmartPtrInputParameter {
```

Comment:
Was this auto generated?

Reply:
No, why ?

Reply:
I'm just curious why this is needed.

Reply:
It was added in https://github.com/apache/arrow/pull/5454/commits/714bb9f169e5b89a58b8b62dd073c4d1cc9163c2. I figured Romain was going somewhere with this so I left it in, but that hasn't happened yet.